File: 120. Triangle/main.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        int length=triangle.size();
        vector<int> dp(triangle.back());
        int i,j;
        //for(i=0;i<length;i++)
        //    dp[i]=triangle[length-1][i];
        for(i=length-1-1;i>=0;i--){
            for(j=0;j<=i;j++){
                dp[j]=triangle[i][j] + min(dp[j],dp[j+1]);
            }
        }
        return dp[0];
    }
};

File: 120. Triangle/main.py
class Solution(object):
    def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        dp = triangle[-1] 
        n = len(triangle)
        
        for i in range(n-2, -1, -1):
            l = triangle[i] 
            for j in range(len(l)):
                dp[j] = l[j] + min(dp[j], dp[j+1]) 
                
        return dp[0]
